src/main/java/it/pagopa/pn/delivery/middleware/notificationdao/CassandraNotificationDao.java

```diff
@@ -0,0 +1,52 @@
+package it.pagopa.pn.delivery.middleware.notificationdao;
+
+import it.pagopa.pn.api.dto.notification.Notification;
+import it.pagopa.pn.commons.abstractions.IdConflictException;
+import it.pagopa.pn.commons.abstractions.KeyValueStore;
+import it.pagopa.pn.commons.abstractions.impl.MiddlewareTypes;
+import it.pagopa.pn.delivery.middleware.NotificationDao;
+import it.pagopa.pn.delivery.model.notification.cassandra.NotificationEntity;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+@Component
+@ConditionalOnProperty( name = NotificationDao.IMPLEMENTATION_TYPE_PROPERTY_NAME, havingValue = MiddlewareTypes.CASSANDRA )
+@Slf4j
+public class CassandraNotificationDao implements NotificationDao {
+
+    private final KeyValueStore<String, NotificationEntity> notificationEntityDao;
+    private final DtoToEntityMapper dto2entityMapper;
+    private final EntityToDtoMapper entity2dtoMapper;
+
+    public CassandraNotificationDao( KeyValueStore<String, NotificationEntity> notificationEntityDao, DtoToEntityMapper dto2entityMapper, EntityToDtoMapper entity2dtoMapper) {
+        this.notificationEntityDao = notificationEntityDao;
+        this.dto2entityMapper = dto2entityMapper;
+        this.entity2dtoMapper = entity2dtoMapper;
+    }
+
+    @Override
+    public void addNotification(Notification notification) throws IdConflictException {
+
+        NotificationEntity entity = dto2entityMapper.dto2Entity( notification );
+        notificationEntityDao.putIfAbsent( entity );
+    }
+
+    @Override
+    public Notification getNotificationByIun(String iun) {
```

Comment:
**Contratto**
Se un metodo o funzione puo' ritornare ``null`` è necessario che sia chiaro dal contratto in uno dei due modi (cosi' che chi usa la API lo sappia di certo ed eviti NPE):
1. usare ``Optional<>``
2. annotare il metodo come ``@Nullable``

Consiglio (1).